### DISCUSSION TITLE
ARM64 Linux VM stuck: "EFI stub: Exiting boot services and installing virtual address map..."

### DISCUSSION THREAD

**Discussion by eveli909** [utmapp/UTM#3743] [UTM Help (macOS)] [answered]
After rebooting UTM virtual machine it was not possible to start VM again.
Tried to change Display setting to Console only, but that doesn't help.
Tried to Run:

`fsck /dev/mapper/ubuntu--vg-ubuntu--lv
` 
However as that is mounted and impossible to umount, it is unable to perform the operation. Also tried to boot the VM from USB, but that leads to the same problem.

UTM version 3.0.4
macOS Monterey

Is there any solution to successfully restart the VM?

**Comment by AndreaPra** [comment]
Same problem...

I see GRUB, type enter on Ubuntu but it ives me this message

**Reply by antoinedarche** [reply]
Same problem...

**Reply by ghost** [reply]
<img width="800" alt="Screen Shot 2022-03-10 at 7 24 16 PM" src="https://user-images.githubusercontent.com/39442069/157652899-63fe335a-e46f-4f2d-ab23-e19e9cd3aa29.png">
<img width="800" alt="image" src="https://user-images.githubusercontent.com/39442069/157652828-1617c014-a79c-40e9-b30f-7e42a878408f.png">


Same problem. I've also tried creating new VMs (Ubuntu 20.04 and 21.10) and updating the UTM app, but I still get the same issue.

**Reply by dhirajftcsolar** [reply]
same problem with UTM 3.1, M1 MACbookpro 2021, Ubuntu 20.04 lts

**Comment by lindacuster** [comment]
Same issue. Sticks after this.

(Note that alignment on 64k boundary note is still present too)


<img width="1054" alt="Screen Shot 2022-03-10 at 9 30 20 AM" src="https://user-images.githubusercontent.com/15956967/157696851-b7ddd969-4352-4abf-8527-e8db169080dd.png">

**Reply by whatboxapp** [reply]
Same problem

**Comment by osy** [comment, marked as answer]
https://github.com/utmapp/UTM/issues/2682#issuecomment-1064827650

**Reply by dhirajpatra** [reply]
This is also not solving the issue. Kindly give the correct steps to create Ubuntu 20.04 LTS VM inside M1 MAC book pro 2021

**Reply by osy** [reply]
Update to 3.1.5

**Reply by KJordanBerg** [reply]
If it helps anyone, I had the issue in this thread today. Updating to 3.1.5 via the app store resolved the issue. Thanks for the prompt UTM update!

**Reply by dhirajpatra** [reply]
Yes it helped thanks

**Comment by freebrowser1** [comment]
That indeed helped: with 3.1.5 it started and I could install xfce4.

Then I installed `sudo apt install spice-vdagent spice-webdavd` but after a reboot of the VM was unable to mount.davfs and could not install it.

**Reply by dhirajpatra** [reply]
Hope this will help https://sleeplessbeastie.eu/2017/09/04/how-to-mount-webdav-share/

**Comment by freebrowser1** [comment]
When I started the VM later again it gets stuck again at (UTM version 3.15):
````
EFI stub: Booting Linux Kernel...
EFI stub: Using DIB from configuration table
EFI stub: Exiting boot services and installing virtual address map...
````

**Reply by dhirajpatra** [reply]
Remove all, restart system and again install UTM 3.15 min version.

**Comment by freebrowser1** [comment]
I did that and it now worked.

The webdav mount did work as well, as I found the page you linked, I had to install `sudo apt install davfs2` on the Ubuntu client.
Still unclear is why I always have to move to *Ubuntu (second option) in the boot screen as "Install Ubuntu" is on top.
When I move the installer ISO from top to below the Ubuntu image, it issues an error.

**Comment by Behinder** [comment]
Tried this with gentoo on UTM 4.4.5 same error appears.

**Comment by soysaucewaso** [comment]
For more modern versions of UTM, what worked for me is in Ubuntu Settings clicking the new under sound -> serial. Next boot into recovery mode, choose root shell.
The command `shutdown -rF now` runs a fsck after unmounting.
